`uv publish` giving `Missing credentials for https://upload.pypi.org/legacy/`

I moved from this GitHub Actions that publishes to PyPI using:

```yaml
      - uses: pypa/gh-action-pypi-publish@release/v1
        with:
          password: ${{ secrets.PYPI_API_TOKEN }}
```

To this with `uv==0.4.20`:

```yaml
      - run: uv publish
        env:
          UV_PUBLISH_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
```

However, `uv publish` fails with:

```none
warning: `uv publish` is experimental and may change without warning
Publishing 2 files https://upload.pypi.org/legacy/
Uploading ldp-0.10.0-py3-none-any.whl (90.7KiB)
error: Failed to publish `dist/repo-0.10.0-py3-none-any.whl` to https://upload.pypi.org/legacy/
  Caused by: Failed to send POST request
  Caused by: Missing credentials for https://upload.pypi.org/legacy/
Error: Process completed with exit code 2.
```

What am I doing wrong here? Also, I am thinking perhaps the stack could be made more user friendly

--- a/crates/uv/src/commands/publish.rs
+++ b/crates/uv/src/commands/publish.rs
@@ -75,6 +75,10 @@ pub(crate) async fn publish(
         (username, password)
     };
 
+    if password.is_some() && username.is_none() {
+        bail!("You need to provide a username with a password, use `--token` for tokens");
+    }
+
     for (file, filename) in files {
         let size = fs_err::metadata(&file)?.len();
         let (bytes, unit) = human_readable_bytes(size);

